app/mobile-content/pages/shared/page-input/page-input.component.ts

```diff
@@ -0,0 +1,160 @@
+  @Input() page: MobilePage;
+  audioTitle = '';
+  imageTitle = '';
+  infoPages = new Map<number, MobilePage>();
+  sliderTitles = new Map<number, string>();
+
+  fontOptions = MobilePage.fontFamilies;
+  pageTypes = MobilePage.pageTypeValues;
+  statusOptions = Status.getValues();
+
+  private selectMediumDialogRef: MdDialogRef<SelectMediumDialogComponent>;
+  private selectPageDialogRef: MdDialogRef<SelectPageDialogComponent>;
+
+  constructor(private dialog: MdDialog,
+              private mediaService: MediaService,
+              private pageService: MobilePageService) {}
+
+  ngOnInit() {
+    if (this.page.audio != null) {
+      this.mediaService.getMediaById(this.page.audio)
+        .then(
+          audio => this.audioTitle = audio.title
+        ).catch();
+    }
+
+    if (this.page.image != null) {
+      this.mediaService.getMediaById(this.page.image)
+        .then(
+          image => this.imageTitle = image.title
+        ).catch();
+    }
+
+    if (this.page.hasInfoPages()) {
+      Promise.all(this.page.additionalInformationPages.map(pageId => this.pageService.getPage(pageId)))
+        .then(
+          (pages: MobilePage[]) => pages.forEach(page => this.infoPages.set(page.id, page))
+        ).catch();
+    }
+
+    if (this.page.images && this.page.images.length > 0) {
+      Promise.all(this.page.images.map(img => this.mediaService.getMediaById(img.image)))
+        .then(
+          (images: Medium[]) => images.forEach(img => this.sliderTitles.set(img.id, img.title))
+        ).catch();
+    }
+  }
+
+  addInfoPage() {
+    this.selectPageDialogRef = this.dialog.open(SelectPageDialogComponent, {
+      width: '75%',
+      data: {
+        asInfoPage: true,
+        excludeIds: [this.page.id, ...this.page.additionalInformationPages]
+      }
+    });
+    this.selectPageDialogRef.afterClosed().subscribe(
+      (selectedPage: MobilePage) => {
+        if (!selectedPage) { return; }
+        this.page.additionalInformationPages.push(selectedPage.id);
+        this.infoPages.set(selectedPage.id, selectedPage);
+      }
+    );
+  }
+
+  clearAudio() {
+    this.page.audio = null;
+    this.audioTitle = '';
+  }
+
+  clearImage() {
+    this.page.image = null;
+    this.imageTitle = '';
+  }
+
+  moveImageDown(image: any) {
+    let pos = this.page.images.indexOf(image);
+    if (pos < this.page.images.length) {
+      this.page.images[pos] = this.page.images.splice(pos + 1, 1, this.page.images[pos])[0];
+    }
+  }
+
+  movePageDown(pageId: number) {
+    let infoPageIds = this.page.additionalInformationPages; // shorthand
+    let pos = infoPageIds.indexOf(pageId);
+    if (pos < infoPageIds.length) {
+      infoPageIds[pos] = infoPageIds.splice(pos + 1, 1, infoPageIds[pos])[0];
+    }
+  }
+
+  moveImageUp(image: any) {
+    let pos = this.page.images.indexOf(image);
+    if (pos > 0) {
+      this.page.images[pos] = this.page.images.splice(pos - 1, 1, this.page.images[pos])[0];
```

Comment:
Hah. If not for JS's `splice` method, this would require 3 lines and a temporary variable. Why not use the tools you have, right? But yeah, this basically just swaps two adjacent array elements.